modules/dataquery/jsx/react.app.js

```diff
@@ -440,6 +440,26 @@ var DataQueryApp = React.createClass({
                 loading: false
             }
         });
+        for (var i = 0; i < fieldsList.length; i++) {
```

Comment:
@jacobpenny I would agree that `foreach` would be a better choice if the code within the block was syncronise, but since all the loop does is make async calls it seems to be an added layer of complexity that is not needed